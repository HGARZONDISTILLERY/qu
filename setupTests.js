import { QueryClient, QueryClientProvider } from 'react-query';
import { render } from '@testing-library/react';

const queryClient = new QueryClient();

const customRender = (ui, options) =>
  render(ui, { wrapper: ({ children }) => <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>, ...options });

export * from '@testing-library/react';
export { customRender as render };
