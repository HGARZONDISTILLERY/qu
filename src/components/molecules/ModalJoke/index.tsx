import { FC } from 'react';
import { Button, Dialog, DialogActions, DialogContent, DialogContentText, Typography, useMediaQuery, useTheme } from '@mui/material';

interface ModalJokeProps {
  showDadJoke: boolean;
  dadJokeIsLoading: boolean;
  renderError: (error: any) => React.ReactNode;
  dadJokeError?: any;
  open: boolean;
  handleClose: () => void;
  dadJokeData?: { joke: string };
  dadJokeRefetch: () => void;
}

const ModalJoke: FC<ModalJokeProps> = ({
  showDadJoke,
  dadJokeIsLoading,
  renderError,
  dadJokeError,
  open,
  handleClose,
  dadJokeData,
  dadJokeRefetch,
}) => {
  const theme = useTheme();
  const fullScreen = useMediaQuery(theme.breakpoints.down('md'));

  return (
    <>
      {showDadJoke && !dadJokeIsLoading && (
        <>
          {renderError(dadJokeError)}
          <Dialog
            fullScreen={fullScreen}
            open={open}
            onClose={handleClose}
            aria-labelledby="responsive-dialog-title"
          >
            <DialogContent sx={{width: '400px'}}>
              <DialogContentText>
                  <strong>{dadJokeData?.joke}</strong>
              </DialogContentText>
              <Typography 
                  sx={{
                    textAlign: 'center',
                    marginTop: '15px',
                    color: 'gray'
                  }}
                  component="h4">
                    Purum pum pshhh!!!!
                </Typography>
            </DialogContent>
            <DialogActions>
              <Button autoFocus onClick={handleClose}>
                Close
              </Button>
              <Button autoFocus onClick={dadJokeRefetch}>
                Refresh
              </Button>
            </DialogActions>
          </Dialog>
        </>
      )}
    </>
  );
};

export default ModalJoke;