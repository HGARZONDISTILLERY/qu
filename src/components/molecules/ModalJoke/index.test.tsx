import ModalJoke from '.';
import {render} from '../../../../setupTests'

test('ModalJoke component renders correctly', () => {
  const joke = {
    "id": "R7UfaahVfFd",
    "joke": "My dog used to chase people on a bike a lot. It got so bad I had to take his bike away.",
    "status": 200
  }

  const tree = render(

    <ModalJoke
      showDadJoke
      dadJokeIsLoading={false}
      renderError={() => <>Error</>}
      dadJokeError={'error'}
      open
      handleClose={() => console.log('close')}
      dadJokeData={joke}
      dadJokeRefetch={() => console.log('refetch')} />
      );
  expect(tree).toMatchSnapshot();
});