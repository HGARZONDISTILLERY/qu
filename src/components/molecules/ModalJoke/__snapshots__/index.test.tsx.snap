// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalJoke component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      Error
    </div>
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="responsive-dialog-title"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="MuiDialogContent-root css-1msgct5-MuiDialogContent-root"
          >
            <p
              class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
            >
              <strong>
                My dog used to chase people on a bike a lot. It got so bad I had to take his bike away.
              </strong>
            </p>
            <h4
              class="MuiTypography-root MuiTypography-body1 css-1y8uj38-MuiTypography-root"
            >
              Purum pum pshhh!!!!
            </h4>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Close
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Refresh
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    Error
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
